db/migrate/20231107110809_undiscard_incorrectly_deleted_events.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class UndiscardIncorrectlyDeletedEvents < ActiveRecord::Migration[7.0]
+  def change
+    reversible do |dir|
+      dir.up do
+        execute <<-SQL
+          WITH discarded_events AS (
+            SELECT events.id AS event_id
+            FROM events
+            INNER JOIN subscriptions ON subscriptions.id = events.subscription_id
+            INNER JOIN plans ON plans.id = subscriptions.plan_id
+            INNER JOIN billable_metrics ON billable_metrics.code = events.code
+            WHERE (events.timestamp::timestamp(0) >= '2023-10-06')
+              AND events.deleted_at IS NOT NULL
+              AND events.deleted_at > billable_metrics.created_at
+              AND billable_metrics.deleted_at IS NULL
+              AND subscriptions.status IN (0, 1) -- pending and active subscriptions
+              AND plans.interval IN (0, 1) -- weekly and monthly intervals (there is no quarterly and yearly subs)
```

Comment:
we don't really know for OSS 😬

Reply:
Good catch!

Reply:
Oh yes 👍 Hmm if we remove this condition probably we also have to take into account all events from the beginning of the year (if there is some yearly plan in OSS)